charts/argo/values.yaml

```diff
@@ -322,9 +322,207 @@ artifactRepository:
+      # subPath: /data
+      ## Use this field to provide a list of Secrets with external certificates. This can be used to to configure
+      ## TLS for MinIO Tenant pods. Create secrets as explained here:
+      ## https://github.com/minio/minio/tree/master/docs/tls/kubernetes#2-create-kubernetes-secret
+      # externalCertSecret:
+      #   - name: tls-ssl-minio
+      #     type: kubernetes.io/tls
+      ## Enable automatic Kubernetes based certificate generation and signing as explained in
+      ## https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster
+      requestAutoCert: true
+      ## Enable S3 specific features such as Bucket DNS which would allow `buckets` to be
+      ## accessible as DNS entries of form `<bucketname>.minio.default.svc.cluster.local`
+      s3:
+        ## This feature is turned off by default
+        bucketDNS: false
+      ## This field is used only when "requestAutoCert" is set to true. Use this field to set CommonName
+      ## for the auto-generated certificate. Internal DNS name for the pod will be used if CommonName is
+      ## not provided. DNS name format is *.minio.default.svc.cluster.local
+      certConfig:
+        commonName: ""
+        organizationName: []
+        dnsNames: []
+      ## PodManagement policy for MinIO Tenant Pods. Can be "OrderedReady" or "Parallel"
+      ## Refer https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#pod-management-policy
+      ## for details.
+      podManagementPolicy: Parallel
+      ## serviceMetadata allows passing additional labels and annotations to MinIO and Console specific
+      ## services created by the operator.
+      serviceMetadata:
+        minioServiceLabels:
+          label: minio-svc
+        minioServiceAnnotations:
+          v2.min.io: minio-svc
+        consoleServiceLabels:
+          label: console-svc
+        consoleServiceAnnotations:
+          v2.min.io: console-svc
+      ## Add environment variables to be set in MinIO container (https://github.com/minio/minio/tree/master/docs/config)
+      # env:
+      # - name: MINIO_BROWSER
+      #   value: "off" # to turn-off browser
+      # - name: MINIO_STORAGE_CLASS_STANDARD
+      #   value: "EC:2"
+      ## PriorityClassName indicates the Pod priority and hence importance of a Pod relative to other Pods.
+      ## This is applied to MinIO pods only.
+      ## Refer Kubernetes documentation for details https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/#priorityclass/
+      # priorityClassName: high-priority
+      ## Define configuration for Console (Graphical user interface for MinIO)
+      ## Refer https://github.com/minio/console
+      console:
+        image: minio/console:v0.6.3
+        replicas: 2
+        consoleSecret:
+          name: console-secret
+        securityContext:
+          runAsUser: 1000
+          runAsGroup: 2000
+          runAsNonRoot: true
```

Comment:
What I do not like here is that `resources` is a list. When we need to customize _(and we do need this)_ it from external _(eighter `--set` or via `--values`)_, we need to define the whole block again since lists cannot be merged...